### PR TITLE
feat(feedback): Add a mailto link, and make it easy to copy an email address in Feedback Details

### PR DESCRIPTION
**Pull request by ryan953**
There's a new send-email button that uses a `mailto` to open your email client and let you start typing.
SCR-20240410-nufl

Clicking on the name/email text will highlight all of it in blue (as shown) and copy "Josh Ferge <josh.ferge@sentry.io>" onto the clipboard:
SCR-20240410-nufy

It's standard format, mail clients will allow you to paste it. It's nice because it includes both parts in a well formatted structure (i generally find it's easier to remove extra bits if needed compared to having to copy+paste more often).

Also, this change makes it so if someone types the same thing into the name & email fields we'll only show it once.

Fixes https://github.com/getsentry/team-replay/issues/414

### CODE REVIEW

**Comment by c298lee** [comment]
Can we make it more obvious that you can click to copy to clipboard? Maybe a tooltip or change to the pointer cursor?

**Comment by ryan953** [comment]
> Can we make it more obvious that you can click to copy to clipboard? Maybe a tooltip or change to the pointer cursor?

@c298lee thats a good idea, I'll put it in for sure

File: static/app/components/feedback/list/feedbackListItem.tsx [hunk 1]
```diff
@@ -95,7 +94,9 @@ const FeedbackListItem = forwardRef<HTMLDivElement, Props>(
             </Row>
 
             <TextOverflow style={{gridArea: 'user'}}>
-              <FeedbackItemUsername feedbackIssue={feedbackItem} detailDisplay={false} />
+              <strong>
+                {feedbackItem.metadata.name ?? feedbackItem.metadata.contact_email}
```
Comment (michellewzhang) [inline static/app/components/feedback/list/feedbackListItem.tsx:98]: might need an additional fallback to `Anonymous User` here:
SCR-20240411-itpe

Comment (ryan953) [reply static/app/components/feedback/list/feedbackListItem.tsx:98]: yeah, i'll write up some tests too.

**Review by michellewzhang — APPROVED** [review]
:shipit:

File: static/app/utils/feedback/types.tsx [hunk 2]
```diff
@@ -10,9 +10,10 @@ export type FeedbackIssue = Overwrite<
     metadata: {
       contact_email: null | string;
       message: string;
-      name: string;
+      name: null | string;
       title: string;
       value: string;
+      initial_priority?: number;
```
Comment (c298lee) [inline static/app/utils/feedback/types.tsx:16]: what's this for?

**Review by c298lee — APPROVED** [review]

File: static/app/utils/feedback/types.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/utils/feedback/types.tsx:16]: needed to make the types work. the real response in the browser has it so i think it was either recently added to issues, or we just missed it when we did the types the first time around.

**Pull request merged**